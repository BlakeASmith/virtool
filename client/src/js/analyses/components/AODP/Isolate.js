import React from "react";
import { map, min, max, mean } from "lodash-es";

import styled from "styled-components";
import { connect } from "react-redux";
import { getActiveHit } from "../../selectors";
import { BoxGroup, Table } from "../../../base";

const StyledTable = styled(Table)`
    text-align: center;

    th {
        text-align: center;
    }
`;

const Round = num => {
    return Math.round(num * 100) / 100;
};

export const IsolateItem = isolate => {
    const components = map(isolate.sequences, sequence => (
        <tr key={sequence.id}>
            <td>{sequence.accession}</td>
            <td>{sequence.hit ? sequence.hit.length : 0} </td>

            <td>{sequence.identities.length ? Round(min(sequence.identities)) : "-"}</td>
            <td>{sequence.identities.length ? Round(mean(sequence.identities)) : "-"}</td>
            <td>{sequence.identities.length ? Round(max(sequence.identities)) : "-"}</td>
        </tr>
    ));

    return (
        <BoxGroup style={{ marginTop: "10px" }}>
            <StyledTable>
                <thead>
                    <tr>
                        <th>Sequence</th>
                        <th>Hits</th>
                        <th>Min</th>
                        <th>Mean</th>
                        <th>Max</th>
                    </tr>
                </thead>
                <tbody>{components}</tbody>
            </StyledTable>
        </BoxGroup>
    );
};

const mapStateToProps = state => {
    const hit = getActiveHit(state);
    return {
        result: hit
    };
};

export default connect(mapStateToProps)(IsolateItem);
