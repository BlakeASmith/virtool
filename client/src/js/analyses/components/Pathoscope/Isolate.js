import { map } from "lodash-es";
import PropTypes from "prop-types";
import React from "react";
import { Flex, FlexItem } from "../../../base/index";
import { toScientificNotation } from "../../../utils/utils";
import Coverage from "./Coverage";

export default class PathoscopeIsolate extends React.Component {
    static propTypes = {
        otuId: PropTypes.string,
        name: PropTypes.string,
        pi: PropTypes.number,
        coverage: PropTypes.number,
        depth: PropTypes.number,
        maxDepth: PropTypes.number,
        reads: PropTypes.number,
        sequences: PropTypes.arrayOf(PropTypes.object),
        setScroll: PropTypes.func,
        showReads: PropTypes.bool
    };

    componentDidMount() {
        this.chartNode.addEventListener("scroll", this.handleScroll);
    }

    componentWillUnmount() {
        this.chartNode.removeEventListener("scroll", this.handleScroll);
    }

    scrollTo = scrollLeft => {
        this.chartNode.scrollLeft = scrollLeft;
    };

    handleScroll = e => {
        this.props.setScroll(this.props.otuId, e.target.scrollLeft);
    };

    render() {
        const chartContainerStyle = {
            overflowX: "scroll",
            marginTop: "5px",
            boxShadow: "inset 0px 0px 3px 1px #dddddd"
        };

        const chartRibbonStyle = {
            whiteSpace: "nowrap"
        };

        const hitComponents = map(this.props.sequences, (hit, i) => (
            <Coverage
                key={i}
                data={hit.align}
                length={hit.length}
                id={hit.id}
                definition={hit.definition}
                yMax={this.props.maxDepth}
                showYAxis={i === 0}
                isolateComponent={this}
            />
        ));

        const piValue = this.props.showReads ? this.props.reads : toScientificNotation(this.props.pi);

        return (
            <div>
                <div className="pathoscope-isolate-header">
                    <Flex>
                        <FlexItem>{this.props.name}</FlexItem>
                        <FlexItem pad={5}>
                            <strong className="small text-success">{piValue}</strong>
                        </FlexItem>
                        <FlexItem pad={5}>
                            <strong className="small text-danger">{this.props.depth.toFixed(1)}</strong>
                        </FlexItem>
                        <FlexItem pad={5}>
                            <strong className="small text-primary">
                                {toScientificNotation(parseFloat(this.props.coverage))}
                            </strong>
                        </FlexItem>
                    </Flex>
                </div>
                <div ref={node => (this.chartNode = node)} style={chartContainerStyle}>
                    <div style={chartRibbonStyle}>{hitComponents}</div>
                </div>
            </div>
        );
    }
}
