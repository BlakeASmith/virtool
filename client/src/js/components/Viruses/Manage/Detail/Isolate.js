/**
 * @license
 * The MIT License (MIT)
 * Copyright 2015 Government of Canada
 *
 * @author
 * Ian Boyes
 *
 * @exports Isolate
 */

import CX from "classnames";
import React from "react";
import { assign } from "lodash-es";
import { Row, Col, Collapse } from "react-bootstrap";
import { Icon, Radio, ListGroupItem } from "virtool/js/components/Base";
import { formatIsolateName } from "virtool/js/utils";

import IsolateForm from "./IsolateForm";

const IsolateHeader = (props) => (
    <h5>
        <Row>
            <Col md={9}>
                {formatIsolateName(props)}
            </Col>
            <Col md={3}>
                {props.children}
            </Col>
        </Row>
    </h5>
);

IsolateHeader.propTypes = {
    sourceType: React.PropTypes.string,
    sourceName: React.PropTypes.string,
    children: React.PropTypes.oneOfType([React.PropTypes.element, React.PropTypes.arrayOf(React.PropTypes.element)])
};


const getInitialState = (props) => ({
    // If no source type is available, "unknown" will be used if restricted source types are enabled otherwise
    // an empty string will be used.
    sourceType: props.sourceType || (props.restrictSourceTypes ? "unknown": ""),
    sourceName: props.sourceName || "",

    pendingRemoval: false,
    collapsed: true,
    editing: false
});

/**
 * An isolate document that is a list item in a list of isolates. Displays the isolate name. Has icon buttons for
 * editing and removing the isolate and setting it as the default isolate for the virus. Can be selected by clicking,
 * which displays the sequences owned by the isolate in a neighbouring panel.
 *
 * @class
 */
export default class Isolate extends React.Component {

    constructor (props) {
        super(props);
        this.state = getInitialState(this.props);
    }

    static propTypes = {
        virusId: React.PropTypes.string.isRequired,
        isolateId: React.PropTypes.string,
        sourceType: React.PropTypes.string,
        sourceName: React.PropTypes.string,
        default: React.PropTypes.bool,

        active: React.PropTypes.bool,
        allowedSourceTypes: React.PropTypes.array,
        restrictSourceTypes: React.PropTypes.bool,
        selectIsolate: React.PropTypes.func,
        canModify: React.PropTypes.bool
    };

    componentDidMount () {
        if (!this.props.isolateId) {
            document.addEventListener("keyup", this.handleKeyUp, true);
        }
    }

    componentWillReceiveProps = (nextProps) => {
        if (nextProps.sourceName !== this.props.sourceName || nextProps.sourceType !== this.props.sourceType) {
            this.setState({
                sourceType: nextProps.sourceType || (this.props.restrictSourceTypes ? "unknown" : ""),
                sourceName: nextProps.sourceName || ""
            });
        }

        if (!this.props.isolateId && nextProps.isolateId) {
            this.componentWillUnmount();
        }
    };

    componentWillUnmount () {
        document.removeEventListener("keyup", this.handleKeyUp, true);
    }

    modalEntered = () => {
        this.setState({
            collapsed: false
        });
    };

    modalExited = () => {
        this.setState({
            collapsed: true
        });
    };

    /**
     * Select the isolate as long as it is not disabled and not already the active isolate. The selection is handled by
     * the parent Isolates component.
     *
     * @param event {object} - The passed event used to prevent the default action.
     * @func
     */
    select = (event) => {
        event.preventDefault();
        if (!this.props.active) {
            this.props.selectIsolate(this.props.isolateId);
        }
    };

    /**
     * Handle a change from the isolate form. Updates state to reflect the current input values.
     *
     * @param changeObject {object} - an object of field values keyed by field names.
     * @func
     */
    handleChange = (changeObject) => {
        this.setState(changeObject);
    };

    /**
     * Called when the form is submitted or the saveIcon is clicked. If the sourceName or sourceType have changed, the
     * updated data is sent to the server. When a response is received, the form is closed. Saving with no change in the
     * data with simply close the form.
     *
     * @param event {object} - The passed event. Used for preventing the default action.
     * @func
     */
    save = (event) => {
        event.preventDefault();

        const hadChange = (
            this.state.sourceType !== this.props.sourceType ||
            this.state.sourceName !== this.props.sourceName
        );

        // Only send an update to the server if there has been a change in the sourceType or sourceName.
        if (hadChange) {
            // Set pendingChange so the component is disabled and a spinner icon is displayed.
            this.setState({pendingChange: true}, () => {
                // Construct a replacement isolate object and send it to the server.
                dispatcher.db.viruses.request("upsert_isolate", {
                    _id: this.props.virusId,
                    new: {
                        isolate_id: this.props.isolateId,
                        source_type: this.state.sourceType,
                        source_name: this.state.sourceName
                    }
                }).success(this.toggleEditing);
            });
        }

        // If the data did not change, just close the form and save sending it to the server.
        else {
            this.toggleEditing();
        }
    };

    /**
     * Remove the isolate as long as it"s not disabled. The removal button is only shown if the isolate is the active
     * isolate, so this method can only successfully be called if props.active is true. The removal is handled by the
     * parent Isolate component.
     *
     * @param event {object} - The passed event used to prevent the default action.
     * @func
     */
    remove = (event) => {
        event.preventDefault();

        // First set state to indicate that the isolate is pending removal. Then, send a request to remove the
        // isolate.
        if (this.props.active) {
            this.setState({pendingRemoval: true}, () => {
                dispatcher.db.viruses.request("remove_isolate", {
                    _id: this.props.virusId,
                    isolate_id: this.props.isolateId
                });
            });
        }
    };

    /**
     * Set the isolate as the default isolate. This method can be called by any Isolate component, not just the active
     * one. Setting the default isolate is handled by the parent Isolate component, so the isolate_id must be passed up.
     *
     * @func
     */
    setAsDefault = () => {
        dispatcher.db.viruses.request("set_default_isolate", {
            _id: this.props.virusId,
            isolate_id: this.props.isolateId
        });
    };

    /**
     * Toggles whether the isolate is being edited. Triggered by clicking the edit or cancel icon buttons.
     *
     * @func
     */
    toggleEditing = () => {
        const newState = assign(this.getInitialState(), {editing: !this.state.editing});

        if (this.state.editing) {
            this.setState(newState, () => {
                document.removeEventListener("keyup", this.handleKeyUp, true);
            });
        } else {
            this.setState(newState, () => {
                document.addEventListener("keyup", this.handleKeyUp, true);
            });
        }
    };

    handleKeyUp = (event) => {
        if (event.keyCode === 27) {
            event.stopImmediatePropagation();
            this.toggleEditing();
        }
    };

    render () {

        const itemProps = {
            onClick: this.props.active ? null: this.select,
            disabled: this.state.pendingRemoval,
            allowFocus: this.props.active,

            className: CX({
                band: this.props.active
            })
        };

        const itemStyle = {
            background: this.state.editing ? "#fcf8e3": null,
            transition: "0.7s background"
        };

        let icons;

        if (this.props.canModify) {
            icons = (
                <div className="icon-group">
                    {this.props.active && !this.state.editing ? (
                        <Icon
                            name="pencil"
                            bsStyle="warning"
                            onClick={this.toggleEditing}
                        />
                    ): null}
                    {this.props.active && !this.state.editing ? (
                        <Icon
                            name="remove"
                            bsStyle="danger"
                            pending={this.state.pendingRemoval}
                            onClick={this.remove}
                        />
                    ): null}
                    {this.state.editing ? (
                        <Icon
                            name="floppy"
                            bsStyle="primary"
                            onClick={this.save}
                        />
                    ): null}
                    {this.state.editing ? (
                        <Icon
                            name="cancel-circle"
                            bsStyle="danger"
                            onClick={this.toggleEditing}
                        />
                    ): null}
                    <Radio
                        onClick={this.setAsDefault}
                        checked={this.props.default}
                    />
                </div>
            )

        }

        return (
            <ListGroupItem {...itemProps} style={itemStyle}>
                <IsolateHeader sourceType={this.state.sourceType} sourceName={this.state.sourceName}>
                    {icons}
                </IsolateHeader>
                <Collapse
                    in={this.props.canModify && this.state.editing}
                    onExited={this.modalExited}
                    onEntered={this.modalEntered}
                >
                    <div>
                        <div style={{height: "15px"}} />
                        <IsolateForm
                            sourceType={this.state.sourceType}
                            sourceName={this.state.sourceName}
                            allowedSourceTypes={this.props.allowedSourceTypes}
                            restrictSourceTypes={this.props.restrictSourceTypes}
                            onChange={this.handleChange}
                            onSubmit={this.save}
                        />
                    </div>
                </Collapse>
            </ListGroupItem>
        );
    }
}
