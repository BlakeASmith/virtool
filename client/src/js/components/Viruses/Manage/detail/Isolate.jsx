/**
 * @license
 * The MIT License (MIT)
 * Copyright 2015 Government of Canada
 *
 * @author
 * Ian Boyes
 *
 * @exports Isolate
 */

'use strict';

var CX = require('classnames');
var React = require('react');
var Row = require('react-bootstrap/lib/Row');
var Col = require('react-bootstrap/lib/Col');

var Icon = require('virtool/js/components/Base/Icon.jsx');
var Radio = require('virtool/js/components/Base/Radio.jsx');
var ListGroupItem = require('virtool/js/components/Base/PushListGroupItem.jsx');

var IsolateForm = require('./IsolateForm.jsx');
var IsolateHeader = require('./IsolateHeader.jsx');

/**
 * An isolate document that is a list item in a list of isolates. Displays the isolate name. Has icon buttons for editing
 * and removing the isolate and setting it as the default isolate for the virus. Can be selected by clicking, which
 * displays the sequences owned by the isolate in a neighbouring panel.
 *
 * @class
 */
var Isolate = React.createClass({

    propTypes: {
        virusId: React.PropTypes.string.isRequired,
        isolateId: React.PropTypes.string,
        sourceType: React.PropTypes.string,
        sourceName: React.PropTypes.string,
        default: React.PropTypes.bool,

        active: React.PropTypes.bool,
        allowedSourceTypes: React.PropTypes.array,
        restrictSourceTypes: React.PropTypes.bool,
        selectIsolate: React.PropTypes.func,
        canModify: React.PropTypes.bool
    },

    getInitialState: function () {
        return {
            // If no source type is available, 'unknown' will be used if restricted source types are enabled otherwise
            // an empty string will be used.
            sourceType: this.props.sourceType || (this.props.restrictSourceTypes ? 'unknown': ''),
            sourceName: this.props.sourceName || '',

            pendingRemoval: false,
            editing: false
        };
    },

    componentDidMount: function () {
        if (!this.props.isolateId) document.addEventListener("keyup", this.handleKeyUp, true);
    },

    componentWillReceiveProps: function (nextProps) {
        if (nextProps.sourceName !== this.props.sourceName || nextProps.sourceType !== this.props.sourceType) {
            this.setState({
                sourceType: nextProps.sourceType || (this.props.restrictSourceTypes ? 'unknown' : ''),
                sourceName: nextProps.sourceName || ''
            });
        }

        if (!this.props.isolateId && nextProps.isolateId) {
            this.componentWillUnmount();
        }
    },

    componentWillUnmount: function () {
        document.removeEventListener("keyup", this.handleKeyUp, true);
    },

    /**
     * Select the isolate as long as it is not disabled and not already the active isolate. The selection is handled by
     * the parent Isolates component.
     *
     * @param event {object} - The passed event used to prevent the default action.
     * @func
     */
    select: function (event) {
        event.preventDefault();
        if (!this.props.active) this.props.selectIsolate(this.props.isolateId);
    },

    /**
     * Handle a change from the isolate form. Updates state to reflect the current input values.
     *
     * @param changeObject {object} - an object of field values keyed by field names.
     * @func
     */
    handleChange: function (changeObject) {
        this.setState(changeObject);
    },

    /**
     * Called when the form is submitted or the saveIcon is clicked. If the sourceName or sourceType have changed, the
     * updated data is sent to the server. When a response is received, the form is closed. Saving with no change in the
     * data with simply close the form.
     *
     * @param event {object} - The passed event. Used for preventing the default action.
     * @func
     */
    save: function (event) {
        event.preventDefault();

        var adding = !this.props.isolateId;

        var hadChange = (
            this.state.sourceType !== this.props.sourceType ||
            this.state.sourceName !== this.props.sourceName
        );

        // Only send an update to the server if there has been a change in the sourceType or sourceName.
        if (hadChange) {
            // Set pendingChange so the component is disabled and a spinner icon is displayed.
            this.setState({pendingChange: true}, function () {
                // Construct a replacement isolate object and send it to the server.
                dispatcher.db.viruses.request('upsert_isolate', {
                    _id: this.props.virusId,
                    new: {
                        isolate_id: this.props.isolateId,
                        source_type: this.state.sourceType,
                        source_name: this.state.sourceName
                    }
                }).success(adding ? this.props.onAdd: this.toggleEditing);
            });
        }

        // If the data didn't change, just close the form and save sending it to the server.
        else {
            this.toggleEditing();
        }
    },

    /**
     * Remove the isolate as long as it's not disabled. The removal button is only shown if the isolate is the active
     * isolate, so this method can only successfully be called if props.active is true. The removal is handled by the
     * parent Isolate component.
     *
     * @param event {object} - The passed event used to prevent the default action.
     * @func
     */
    remove: function (event) {
        event.preventDefault();

        // First set state to indicate that the isolate is pending removal. Then, send a request to remove the
        // isolate.
        if (this.props.active) this.setState({pendingRemoval: true}, function () {
            dispatcher.db.viruses.request('remove_isolate', {
                _id: this.props.virusId,
                isolate_id: this.props.isolateId
            });
        });
    },

    /**
     * Set the isolate as the default isolate. This method can be called by any Isolate component, not just the active
     * one. Setting the default isolate is handled by the parent Isolate component, so the isolate_id must be passed up.
     *
     * @func
     */
    setAsDefault: function () {
        dispatcher.db.viruses.request('set_default_isolate', {
            _id: this.props.virusId,
            isolate_id: this.props.isolateId
        }, this.onSuccess);
    },

    /**
     * Toggles whether the isolate is being edited. Triggered by clicking the edit or cancel icon buttons.
     *
     * @func
     */
    toggleEditing: function () {
        var newState = _.assign(this.getInitialState(), {editing: !this.state.editing});

        if (this.state.editing) {
            this.setState(newState, function () {
                document.removeEventListener("keyup", this.handleKeyUp, true);
            });
        } else {
            this.setState(newState, function () {
                document.addEventListener("keyup", this.handleKeyUp, true);
            });
        }
    },

    handleKeyUp: function (event) {
        if (event.keyCode === 27) {
            event.stopImmediatePropagation();
            this.props.isolateId ? this.toggleEditing(): this.props.onAdd();
        }
    },

    render: function () {

        var adding = !this.props.isolateId;

        var icons;

        var isolateForm;

        // If the Isolate component is editing or adding an isolate document, show a form and special icon buttons for
        // saving an clearing changes.
        if (this.props.canModify) {
            if (adding || this.state.editing) {
                icons = (
                    <div className='icon-group'>
                        <Icon
                            name='floppy'
                            bsStyle='primary'
                            onClick={this.save}
                        />
                        <Icon
                            name='cancel-circle'
                            bsStyle='danger'
                            onClick={adding ? this.props.onAdd: this.toggleEditing}
                        />
                    </div>
                );

                isolateForm = (
                    <div style={{paddingTop: '10px'}}>
                        <IsolateForm
                            sourceType={this.state.sourceType}
                            sourceName={this.state.sourceName}
                            allowedSourceTypes={this.props.allowedSourceTypes}
                            restrictSourceTypes={this.props.restrictSourceTypes}
                            onChange={this.handleChange}
                            onSubmit={this.save}
                        />
                    </div>
                );
            } else {
                // Set icons to remove and edit the isolate document.
                var removeIcon = this.props.active ? (
                    <Icon
                        name='remove'
                        pending={this.state.pendingRemoval}
                        onClick={this.remove}
                        bsStyle='danger'
                    />
                ) : null;

                icons = (
                    <div className='icon-group'>
                        {this.props.active ?
                            <Icon name='pencil' onClick={this.toggleEditing} bsStyle='warning'/> : null}
                        {this.props.active && !this.state.pendingRemoval ? removeIcon : null}
                        {!this.state.pendingRemoval ?
                            <Radio onClick={this.setAsDefault} checked={this.props.default}/> : null}
                    </div>
                );
            }
        }

        var itemProps = {
            onClick: this.props.active ? null: this.select,
            disabled: this.state.pendingRemoval,
            allowFocus: this.props.active
        };

        // Classes to apply to the ListGroupItem.
        itemProps.className = CX({
            band: this.props.active,
            primary: adding,
            warning: this.state.editing && !adding
        });

        return (
            <ListGroupItem {...itemProps}>
                <IsolateHeader sourceType={this.state.sourceType} sourceName={this.state.sourceName}>
                    {icons}
                </IsolateHeader>
                {isolateForm}
            </ListGroupItem>
        );
    }
});

module.exports = Isolate;