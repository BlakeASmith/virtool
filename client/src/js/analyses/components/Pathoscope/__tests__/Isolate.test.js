import { PathoscopeIsolate, PathoscopeIsolateWeight } from "../Isolate";

describe("<PathoscopeIsolate />", () => {
    it("should render", () => {
        const props = {
            depth: 13,
            maxDepth: 22,
            name: "Isolate A",
            sequences: [
                { filled: [1, 2, 3, 4, 5], length: 5, id: "foo", definition: "Foo Hit" },
                { filled: [1, 2, 3, 4, 5, 4, 2, 3], length: 8, id: "bar", definition: "Bar Hit" }
            ],
            reads: 102
        };
        const wrapper = shallow(<PathoscopeIsolate {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("<PathoscopeIsolateWeight />", () => {
    let props;

    beforeEach(() => {
        props = { pi: 0.321, reads: 121, showReads: false };
    });

    it("should render weight", () => {
        const wrapper = shallow(<PathoscopeIsolateWeight {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render weight in scientific notation", () => {
        props.pi = 0.000000321;
        const wrapper = shallow(<PathoscopeIsolateWeight {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render reads", () => {
        props.showReads = true;
        const wrapper = shallow(<PathoscopeIsolateWeight {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
