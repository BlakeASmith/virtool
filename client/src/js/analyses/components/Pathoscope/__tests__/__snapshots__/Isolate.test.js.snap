// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PathoscopeIsolate /> should render 1`] = `
<Isolate__StyledPathoscopeIsolate>
  <Isolate__PathoscopeIsolateHeader>
    Isolate A
    <PathoscopeIsolateWeight
      reads={102}
    />
    <strong
      className="small text-danger"
    >
      13
    </strong>
    <strong
      className="small text-primary"
    >
      NaN
    </strong>
  </Isolate__PathoscopeIsolateHeader>
  <ScrollSyncPane
    enabled={true}
    group="default"
  >
    <Isolate__PathoscopeChartRibbon>
      <CoverageChart
        definition="Foo Hit"
        id="foo"
        key="0"
        length={5}
        showYAxis={true}
        yMax={22}
      />
      <CoverageChart
        definition="Bar Hit"
        id="bar"
        key="1"
        length={8}
        showYAxis={false}
        yMax={22}
      />
    </Isolate__PathoscopeChartRibbon>
  </ScrollSyncPane>
</Isolate__StyledPathoscopeIsolate>
`;

exports[`<PathoscopeIsolateWeight /> should render reads 1`] = `
<strong
  className="small text-success"
>
  121
</strong>
`;

exports[`<PathoscopeIsolateWeight /> should render weight 1`] = `
<strong
  className="small text-success"
>
  0.321
</strong>
`;

exports[`<PathoscopeIsolateWeight /> should render weight in scientific notation 1`] = `
<strong
  className="small text-success"
>
  3.21E-7
</strong>
`;
